### ISSUE TITLE
NNI installation issue with ConfigSpace-0.4.7

### ISSUE THREAD

**Issue by maxxx580** [microsoft/nni#3909]
**Describe the issue**:

`nni[BOHB]` has a dependency `ConfigSpace==0.4.7`, which causes the installation to fail with the error below. The issue is originated from the ConfigSpace package. ConfigSpace uses the `numpy` in its `setup.py` without declare it as a `setup_requires` ([source](https://github.com/automl/ConfigSpace/blob/v0.4.7/setup.py#L4)). If `numpy` is not installed in the current environment, the installation will fail. 

```
Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-t78p1yeg/configspace_3b56d223c52b4c88851d1ac2a6c0dfa1/setup.py", line 4, in <module>
        import numpy as np
    ModuleNotFoundError: No module named 'numpy'
```

This problem has been addressed in `ConfigSpace==0.4.11` and onwards. I am wondering if NNI could upgrade ConfigSpace to this version or later to avoid the installation error. thanks. 

**Environment**:
- NNI version: 2.2 with BOHB
- Training service (local|remote|pai|aml|etc):local
- Client OS: linux
- Server OS (for remote mode only): linux
- Python version: 3.7
- PyTorch/TensorFlow version: 1.8
- Is conda/virtualenv/venv used?: yes
- Is running in Docker?: no

**Configuration**:
 - Experiment config (remember to remove secrets!): N/A
 - Search space: N/A

**Log message**:
 - nnimanager.log: N/A
 - dispatcher.log: N/A
 - nnictl stdout and stderr: N/A
 

**How to reproduce it?**:
1. create an empty environment without `numpy`
2. run `pip install nni[BOHB]`

**Comment by nhopp94** [comment]
I'm also encountering this. Did you find any workarounds?

**Comment by liuzhe-lz** [comment]
Fixed in v2.5.
